client/components/FailedGraph.vue

```diff
@@ -33,61 +37,72 @@
+        var totalCount = 0; 
         if (this.statshistory && this.stats){
           this.datacollection = {};
           this.datacollection["labels"]= this.statshistory.map(eachStatistic => "" );
           this.datacollection["datasets"]=[];
           for (var thisPath in this.stats.FunctionStatsMap){
-            var dataSetForPath = {
-              label: thisPath + ": "+ this.stats.FunctionStatsMap[thisPath].Failed,
-              fill: true,                                       // Set to true because chart is stacked
-              backgroundColor: getBackgroundColorFor(thisPath), // Set color because chart is stacked
-              borderColor: getBorderColorFor(thisPath),
-              borderWidth: lineWidthInPixels,
-              radius:pointRadiusInPixels,
-              data: this.statshistory.map(eachStatistic => eachStatistic.FunctionStatsMap[thisPath].Failed)
-            };
-            this.datacollection["datasets"].push(dataSetForPath);
-             
+            if (this.routes==null || isPathIn(thisPath,this.routes)){
+              totalCount = totalCount + this.stats.FunctionStatsMap[thisPath].Failed;          
+              var dataSetForPath = {
+                label: thisPath + ": "+ this.stats.FunctionStatsMap[thisPath].Failed,
+                fill: true,                                       // Set to true because chart is stacked
+                backgroundColor: getBackgroundColorFor(thisPath), // Set color because chart is stacked
+                borderColor: getBorderColorFor(thisPath),
+                borderWidth: lineWidthInPixels,
+                radius:pointRadiusInPixels,
+                data: this.statshistory.map(eachStatistic => eachStatistic.FunctionStatsMap[thisPath].Failed)
+              };
+              this.datacollection["datasets"].push(dataSetForPath);
+            }   
           }
-          var legs = document.getElementById("failedGraphLegend");
-        
+          this.total = totalCount;
+
           // legend  
+          var legs = document.getElementById("failedGraphLegend");
           var text = [];
           text.push('<ul class=\'' + 'chartLegend\'>');
           var chartDataDatasets = this.datacollection["datasets"];
           var chartDataDatasetsLength = chartDataDatasets.length;
-		  for (var i = 0; i < chartDataDatasets.length; i++) {
-			text.push('<li><span class=\'chartLabelEmblem\' style=\'' +
-			  'background-color:' + chartDataDatasets[i].backgroundColor + '; ' +
-		      'border-color:' + chartDataDatasets[i].borderColor + ';' +
-			  '\'></span>');
-			if (chartDataDatasets[i].label) {
-			  text.push('<span class=\'chartLabelText\'>'+chartDataDatasets[i].label+'</span>');
-			}
-			text.push('</li>');
-		  }
-		  text.push('</ul>');
-          legs.innerHTML  = text.join(''); 
+          for (var i = 0; i < chartDataDatasets.length; i++) {
```

Comment:
Common code see https://github.com/fnproject/ui/pull/37/files#diff-a3db016c87a936e401dcbc268b43db7cR92

Reply:
Done. All four graphs now use a shared function in `graphUtilities.js`.